### PR TITLE
Support (read,connect) timeout tuple for `request_timeout` config param

### LINKED ISSUE #440: Split request_timeout config option to read and connect timeout
Needed for #439, but also to allow for finer granularity of timeouts.

We could introduce two new options and/or support a tuple for `request_timeout` (similar to the requests lib).

See also #357.

### PR DESCRIPTION
**Pull request by randomir**
Close #440.

### CODE REVIEW

File: dwave/cloud/config/models.py [hunk 1]
```diff
@@ -99,6 +101,12 @@ class PollingSchedule(BaseModel, GetterMixin):
     backoff_base: Optional[float] = 1.3
 
 
+def _literal_eval(obj):
+    if isinstance(obj, str):
+        return ast.literal_eval(obj)
```
Comment (arcondello) [inline dwave/cloud/config/models.py:106]: Didn't know about this, neat

**Review by arcondello — APPROVED** [review]
LGTM

File: dwave/cloud/config/models.py [hunk 2]
```diff
@@ -99,6 +101,12 @@ class PollingSchedule(BaseModel, GetterMixin):
...
+    return obj
+
```
Comment (JoelPasvolsky) [inline dwave/cloud/config/models.py:108]: Is this a leftover? I don't see it used (plus line 15)

**Review by JoelPasvolsky — APPROVED** [review]

File: dwave/cloud/config/models.py (hunk 2 above)
Comment (randomir) [reply dwave/cloud/config/models.py:108]: it's used in line 140 to define field validator

Comment (JoelPasvolsky) [reply dwave/cloud/config/models.py:108]: Ah, there you go, thanks!

File: dwave/cloud/config/models.py (hunk 1 above)
Comment (randomir) [reply dwave/cloud/config/models.py:106]: Yup, we use it to safely parse linear/quad list/dict in [`dwave sample`](https://github.com/dwavesystems/dwave-cloud-client/blob/68078b1ca19f5534d36b469ac968eaf1c6e6322f/dwave/cloud/cli.py#L606) as well.

**Pull request merged**